From 223b5e8ffb7fde87fd20c2d6fe0f80a22785a9c4 Mon Sep 17 00:00:00 2001
From: "tpopela@redhat.com"
 <tpopela@redhat.com@268f45cc-cd09-0410-ab3c-d52691b4dbfc>
Date: Wed, 15 Mar 2017 05:52:08 +0000
Subject: [PATCH] Wrong condition in offlineasm/risc.rb
 https://bugs.webkit.org/show_bug.cgi?id=169597

Reviewed by Mark Lam.

It's missing the 'and' operator between the conditions.

* offlineasm/risc.rb:

git-svn-id: http://svn.webkit.org/repository/webkit/trunk@213973 268f45cc-cd09-0410-ab3c-d52691b4dbfc
---
 Source/JavaScriptCore/offlineasm/risc.rb |  2 +-
 2 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/Source/JavaScriptCore/offlineasm/risc.rb b/Source/JavaScriptCore/offlineasm/risc.rb
index 3fbc07d..28e0165 100644
--- a/Source/JavaScriptCore/offlineasm/risc.rb
+++ b/Source/JavaScriptCore/offlineasm/risc.rb
@@ -374,7 +374,7 @@ def riscLowerMalformedImmediates(list, validImmediates)
             when "addi", "addp", "addq", "addis", "subi", "subp", "subq", "subis"
                 if node.operands[0].is_a? Immediate and
                         (not validImmediates.include? node.operands[0].value) and
-                        validImmediates.include? -node.operands[0].value
+                        validImmediates.include? -node.operands[0].value and
                         node.operands.size == 2
                     if node.opcode =~ /add/
                         newOpcode = "sub" + $~.post_match
-- 
2.9.3

